react-hot-button-component/src/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const colorClass = [
+  'color-indigo',
+  'color-mediumslateblue',
+  'color-tomato',
+  'color-lightsalmon',
+  'color-gold',
+  'color-white'
+];
+
+class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      clicks: 0
+    };
+  }
+
+  handleClick = () => {
+    this.setState({
+      clicks: this.state.clicks + 1
+    });
+  }
+
+  render() {
+    let index = 0;
+
+    if (this.state.clicks < 3) index = 0;
```

Comment:
While this is a solid solution, I recommend moving the logic for choosing the class into a method which can be called from render and which will return the proper index value